daemon/logger/journald/journald.go

```diff
@@ -81,6 +81,12 @@ func New(info logger.Info) (logger.Logger, error) {
 	for k, v := range extraAttrs {
 		vars[k] = v
 	}
+	if info.ContainerLabels["com.docker.stack.namespace"] != "" {
+		vars["SWARM_STACK"] = info.ContainerLabels["com.docker.stack.namespace"]
+	}
+	if info.ContainerLabels["com.docker.swarm.service.name"] != "" {
+		vars["SWARM_SERVICE"] = info.ContainerLabels["com.docker.swarm.service.name"]
+	}
```

Comment:
I'm a bit on the fence if we should do this (I get where you're coming from though!).

- first of all scope; adding this may result in ending up in a bit of a sliding slope; `docker stack` is just _one_ example of tools using labels to add more metadata, but other tools may as well (for example, `docker compose` also adds labels, so we may end up adding a `COMPOSE_PROJECT` and `COMPOSE_SERVICE` etc etc
- other logging drivers also add some metadata  to structured logs (e.g. the [fluentd](https://docs.docker.com/config/containers/logging/fluentd/); adding the metadata as a default in this driver, may end us up with also having to add it to the other drivers.

So it will be a bit difficult to determine "where to draw the line". Note that some drivers already provide a `labels` and `labels-regex` option, which could probably provide a way to add these labels (include all labels starting with `com.docker.swarm` or `com.docker.stack`). Doing so would keep the logging driver to be less "opinionated".

Reply:
@thaJeztah I agree it's not the best solution. I'm not very experienced with Go, and my main criteria was having the same fields output when either the loki or journald driver is used, since those are the two I use mostly. Agree ideally need something that is consistent across all drivers, or at least one that support outputting arbitrary fields, and that the fluentd approach seems good. On the other hand, also think a ~kludge to support swarm and compose would be OK too, since these seem to be the main cases AFAIK. In fact the loki driver [includes](https://github.com/grafana/loki/blob/5206e3acd0e6c622c54376f6138c881efe1c656e/clients/cmd/docker-driver/config.go#L60) the compose labels you referred to. I think it was an oversight not to add them in this PR. I will add. Still can understand if want to wait for a better solution.